src/main/java/teammates/ui/webapi/action/GetCourseJoinStatusAction.java

```diff
@@ -49,6 +50,13 @@ private JsonResult getInstructorJoinStatus(String regkey) {
         if (instructor == null) {
             return new JsonResult("No instructor with given registration key: " + regkey, HttpStatus.SC_NOT_FOUND);
         }
+
+        String institute = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_INSTITUTION);
+        String mac = getRequestParamValue(Const.ParamsNames.INSTITUTION_MAC);
+        if (institute != null && !StringHelper.isCorrectSignature(institute, mac)) {
+            return new JsonResult("Error in URL validation. Please re-check the URL.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Have you done manual testing to ensure that this is displayed correctly?

Reply:
After manual testing, this does not seem to work as intended. Additionally, neither does the institute validation, the authentication fails even with the correct signature in the url params. 

All back-end checks/tests indicate that it should be working so I believe there might be some disconnect with the front-end. I will investigate this further and update once the implementation works properly.